src/ddmd/expression.d

```diff
@@ -7823,6 +7823,23 @@ extern (C++) class UnaExp : Expression
         return null;
     }
 
+    final Expression incompatibleTypes()
+    {
+        if (e1.type.toBasetype() != Type.terror)
+        {
+            if (e1.op == TOKtype)
+            {
+                error("incompatible type for (%s(%s)): cannot use '%s' with types", Token.toChars(op), e1.toChars(), Token.toChars(op));
+            }
+            else
+            {
+                error("incompatible type for (%s(%s)): '%s'", Token.toChars(op), e1.toChars(), e1.type.toChars());
+            }
+            return new ErrorExp();
+        }
+        return this;
```

Comment:
Hmm, what is this returning? Errors should be propagated, shouldn't it return `new ErrorExp()`?